source/Octopus.Tentacle.Tests.Integration/Util/Builders/Decorators/Proxies/ServiceProxy.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Reflection;
+using System.Runtime.ExceptionServices;
+using System.Threading.Tasks;
+
+namespace Octopus.Tentacle.Tests.Integration.Util.Builders.Decorators.Proxies
+{
+    public abstract class ServiceProxy : DispatchProxyAsync
+    {
+        protected object? TargetService { get; private set; }
+
+        protected void SetTargetService(object service)
+        {
+            TargetService = service;
+        }
+
+        public override object Invoke(MethodInfo method, object[] args)
+        {
+            EnsureTargetServiceNotNull();
+            try
+            {
+                OnStartingInvocation(method);
+
+                return method.Invoke(TargetService, args)!;
+            }
+            catch (TargetInvocationException e)
```

Comment:
If the invocation throws an exception, it gets wrapped in a`TargetInvocationException`, which we then unwrap